tracer/src/Datadog.Trace/Datadog.Trace.csproj

```diff
@@ -11,6 +11,7 @@
     <SymbolPackageFormat>snupkg</SymbolPackageFormat>
     <PackageReadmeFile>README.md</PackageReadmeFile>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    <LangVersion>preview</LangVersion>
```

Comment:
is this intended?

Reply:
I meant to remove it. The solution wouldn't compile for me without it because of the use of some new pattern matching construct. I guess it works from the CLI (and Ryder) because it's officially released in .NET 7.0, which is unsupported in VS2022.

Now I think about perhaps it's better to leave it in to stop the other VS users having to add it repeatedly? I don't have a strong opinion either way.

Reply:
sgtm, I've been having the same problem too in rider in fact

Reply:
I don't think we should use `<LangVersion>preview</LangVersion>` - .NET 7 is fully supported in both VS 2022 and Rider🤔

Reply:
I have this version of rider now:

```
JetBrains Rider 2022.3.1
Build #RD-223.8214.53, built on December 20, 2022

Runtime version: 17.0.5+1-b653.23 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11 10.0
.NET 7.0.0 (Server GC)
```

When I compile without preview I get:
```
  W3CTraceContextPropagator.cs(429, 24): [CS8652] The feature 'pattern matching ReadOnly/Span<char> on constant string' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
```

Any idea what I'm doing wrong? I think code was added by @zacharycmontoya , maybe he knows.

Reply:
removed preview in 7ae1553d3

Reply:
> When I compile without preview...

So it's not an IDE issue 🤔 

> [CS8652] The feature 'pattern matching ReadOnly/Span<char> on constant string' is currently in Preview and *unsupported*.

That's... surprising... I guess it was never merged to C#11, even though it's [listed right there in the docs](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-11.0/pattern-match-span-of-char-on-string). I wonder if that's a bug 🤔